gateway-ha/src/main/java/io/trino/gateway/ha/resource/GatewayWebAppResource.java

```diff
@@ -423,4 +439,71 @@ public Response readExactMatchSourceSelector()
         List<ResourceGroupsManager.ExactSelectorsDetail> selectorsDetailList = resourceGroupsManager.readExactMatchSourceSelector();
         return Response.ok(Result.ok(selectorsDetailList)).build();
     }
+
+    @GET
+    @RolesAllowed("USER")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("/getRoutingRules")
+    public Response getRoutingRules()
+    {
+        try {
+            String rulesConfigPath = configuration.getRoutingRules().getRulesConfigPath();
+            YAMLFactory yamlFactory = new YAMLFactory();
+            ObjectMapper yamlReader = new ObjectMapper(yamlFactory);
+            YAMLParser yamlParser = yamlFactory.createParser(new String(Files.readAllBytes(Paths.get(rulesConfigPath))));
```

Comment:
Please consider moving `ObjectMapper` to fields and initialize it in the constructor.